app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
+
+  def update_posts_counter
+    author.increment!(:posts_counter)
+  end
+
```

Comment:
Kindly here you can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `update_posts_counter` method every time a new post has been saved to database, then make the update_post_counter method private.
Example:
```ruby
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
``` 

Also consider doing the same in the update methods in the other files.